runelite-client/src/main/java/net/runelite/client/util/LinkBrowser.java

```diff
@@ -44,6 +45,7 @@
 public class LinkBrowser
 {
 	private static boolean shouldAttemptXdg = OSType.getOSType() == OSType.Linux;
+	private static boolean isSnapClient = ! Strings.isNullOrEmpty(System.getenv("SNAP"));
```

Comment:
This is not specific to snap only so this check do not makes sense. If you are worried about blocking threads, then just send it to different thread

Reply:
After considering everything, from making the entire class threaded or seperating this class into two, and having looked at the actual use of the existing and new methods, it does appear the best thing to do is just throw the method in an anonymous thread if the threading is a concern. There isn't enough usage of this to begin with to overcomplicate it.